### ISSUE TITLE
Maintenance: delete/rebase old branches

### ISSUE THREAD

**Issue by kyordhel** [RoboCupAtHome/RuleBook#277]
There are quite some old branches that have strongly diverged from *master*. To keep the repo clean, those branches need to be deleted or updated (``git rebase master``).

- [x] ``2017/gpsr``
- [x] ``fix/231``
- [x] ``feature/ip_assignments``
- [x] ``dataloggin_formats``

In addition, it would be a nice idea to define use policies. Some examples can be
- Convention on branch updating: rebase/merge (I think rebase is cleaner).
- Convention on branches
  - dashes- or slashes/
  - year/author/topic
    - year/assignee/#issue
    - year/author/the_feature
  - topic/subtopic
    - fix/#issue
    - feature/the_feature
    - hotfix/the_hotfix
  - Name casing: CamelCasing or under_scoring
  - Expiracy date
- Convention on files
  - Commenting style
  - Orthography standard: US/UK
  - Name casing: CamelCasing or under_scoring or dotted.words
- Others

**Comment by LoyVanBeek** [comment]
I deleted a lot of old/merged branches last weekend but some are left indeed. 

My policy proposal: 

- Branch is deleted after merging. Review should catch initial issues before merging, later stuff is fixed in separate branches
- New challenges: <year>/challenge_name. (e.g. 2017/set_a_table)
- New general rules: feature/description_of_new_stuff. (eg. feature/eegpsr_example_scoresheet, feature/eegpsr_shrink_scoresheet)
- To address issues: fix/#<github issue number>_<few_word_description>.
- No author names in branches, anyone should be able to work on any branch. 
- Rebase local branch onto master before pushing it to GitHub. No rebasing of public history, that messes everybody's stuff up. I always ```git pull --rebase``` to keep history as linear as possible. My work is then applied at the end of anyone else's. 
- When merging a pull request, I want to have traceability, so a merge commit. Certainly not a "Squash and merge" but a rebase and merge could be OK for a small fix.

**Comment by kyordhel** [comment]
All this must be written and signed.

I am more for CamelCase and support your branching naming, more or less like this
- year/TestOrChallengeName
- fix/#NumberIssue
- hotfix/SuperUrgentPatch

Also agree on not naming branches, just try not having two people working on the same branch at the same time.

**Comment by LoyVanBeek** [comment]
I'm not in for a fight about CamelCase vs. much more_readable_names, but what's a good place to put these guidelines?

**Comment by kyordhel** [comment]
I would vote for enabling the wiki of the repo and add a page with the guidelines and conventions.
Regarding the fight (or death match) on camelCase, CamelCase, and more_readable_names, consensus is required. Our votes are set, the rest of the committee needs to give their votes.

**Comment by balkce** [comment]
+1 for enabling the wiki of the repo, we could place there some guidelines for editing the rule book and how the community should interact with the rule book, and we could use as a landing page for new TC members.

In terms of naming conventions, I prefer underscoring since they are more readable; CamelCase screams arrogance =P

**Comment by kyordhel** [comment]
Is good to know we have reached consensus on using camelCase. Of course nothing less can be expected from the Mighty RuleMakers!!!

Now, time for drinking a glass of wine...
![frog](https://cloud.githubusercontent.com/assets/8583949/26218225/89f77232-3c0a-11e7-8c5b-3b3ff0c8cc9d.jpg)

**Comment by LoyVanBeek** [comment]
The wiki is already enabled. I enabled it when my LaTeX wouldn't compile at RoboCup 2015 in Hefei and wanted to document how to fix it.

**Comment by kyordhel** [comment]
I'll try to add something to the wiki then.

**Comment by kyordhel** [comment]
### Update

Now there is a Wiki page with guidelines for [Questions](https://github.com/RoboCupAtHome/RuleBook/wiki/Guidelines:-Questions) and [Issues](https://github.com/RoboCupAtHome/RuleBook/wiki/Guidelines:-Issues), and a very brief [F.A.Q.](https://github.com/RoboCupAtHome/RuleBook/wiki/FAQ:-Frequently-Asked-Questions)

README.md has been updated (#305) to point to these files, with a naive wish of having clearer issues.

Help is required.

**Comment by balkce** [comment]
Seems good to me. In what do you need help?

**Comment by kyordhel** [comment]
Extending the FAQ and the wiki, updating the guidelines, etc.
